velox/functions/sparksql/Rand.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/functions/Macros.h"
+
+namespace facebook::velox::functions {
+
+template <typename T>
+struct RandFunction {
+  static constexpr bool is_deterministic = false;
+
+  std::optional<std::mt19937> generator;
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const core::QueryConfig& /*config*/,
+      const int32_t* seed,
+      const int32_t* partitionIndex) {
+    if (seed == nullptr || partitionIndex == nullptr) {
+      VELOX_USER_FAIL(
+          "Spark rand/random function requires constant seed/partitionIndex!");
+    }
+  }
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const core::QueryConfig& /*config*/,
+      const int64_t* seed,
+      const int32_t* partitionIndex) {
+    if (seed == nullptr || partitionIndex == nullptr) {
+      VELOX_USER_FAIL(
+          "Spark rand/random function requires constant seed/partitionIndex!");
+    }
+  }
+
+  // Partition index is NOT needed to differentiate generator as ThreadLocalPRNG
```

Comment:
Thanks for your review!
This comment is to say each thread can independently produce random numbers without the usage of partitionIndex to differentiate generators. It may be misleading, should I remove it?